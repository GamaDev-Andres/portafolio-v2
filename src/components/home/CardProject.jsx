import { Link } from 'react-router-dom'
const CardProject = ({ project }) => {
  const { poster, descripcion, tecnologias, titulo, url } = project
  return (
    <div className=" shadow-blue-200 shadow-md rounded-2xl bg-white text-fondos">
      <a target="_blank" href={url} className="w-full block aspect-video">
        <img
          className="w-full aspect-video object-cover"
          src={poster}
          alt="project"
        />
      </a>
      <div className="px-4 pb-4 ">
        <a target="_blank" href={url}>
          <h3 className="font-bold text-blue-400 text-center text-xl m-2">
            {titulo}
          </h3>
        </a>
        <div>
          <h3 className="text-letra font-medium">Descripción</h3>

          <p>{descripcion}</p>
        </div>
        <div>
          <h3 className="text-letra font-medium">Tecnologias implementadas</h3>
          <ul className="flex gap-2 flex-wrap py-2">
            {tecnologias.map((el) => (
              <li
                key={el}
                className="px-2 py-1 border border-bordes rounded-full"
              >
                <span>{el}</span>
              </li>
            ))}
          </ul>
        </div>
        <Link
          className=" hover:shadow-lg hover:shadow-blue-300 text-sm text-center mx-auto block mt-4 border border-solid shadow-blue-200 shadow-md border-blue-400 px-6 py-2 rounded-full"
          to="/project/barberapp"
          rel="noreferrer"
        >
          Ver más detalles
        </Link>
      </div>
    </div>
  )
}

export default CardProject
