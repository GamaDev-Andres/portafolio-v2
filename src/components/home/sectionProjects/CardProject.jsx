import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import ListOfTecnologias from './ListOfTecnologias'
import TitleAndLinksProject from './TitleAndLinksProject'
const CardProject = ({ project, withDetails = false }) => {
  const { poster, descripcion, tecnologias, titulo, url } = project
  return (
    <div className=" shadow-blue-200 shadow-md rounded-2xl rounded-t-none transition-colors duration-150 bg-white dark:bg-baseDark text-fondos">
      <a
        target="_blank"
        href={url}
        className="w-full block aspect-video"
        rel="noreferrer"
      >
        <img
          className="w-full aspect-video object-cover"
          src={poster}
          alt="project"
        />
      </a>
      <div className="px-4 pb-4">
        <TitleAndLinksProject project={project} />
        <div>
          <h3 className=" dark:text-letraDark text-letra font-medium">
            Descripción
          </h3>

          <p>{descripcion}</p>
        </div>
        <ListOfTecnologias tecnologias={tecnologias} />
        {withDetails && (
          <Link
            className=" hover:shadow-lg hover:shadow-blue-300 text-sm text-center mx-auto block mt-4 border border-solid shadow-blue-200 shadow-md border-blue-400 px-6 py-2 rounded-full"
            to={`/project/${titulo}`}
            rel="noreferrer"
          >
            Ver más detalles
          </Link>
        )}
      </div>
    </div>
  )
}
CardProject.propTypes = {
  project: PropTypes.object.isRequired,
  withDetails: PropTypes.bool
}
export default CardProject
