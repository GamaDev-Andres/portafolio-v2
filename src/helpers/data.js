export const proyectos = [
  {
    titulo: 'BARBERAPP',
    descripcion: `Aplicación realizada con el stack MERN y styled components,
     enfocada en la administración de una barbería en donde el cliente tiene 
     permitido agendar una cita con un barbero basándose en su perfil ,
      cortes o calificación de otros clientes.`,
    tecnologias: [
      'React',
      'Express',
      'StyledComponents',
      'MongoDb',
      'Reducers'
    ],
    url: 'https://barberappag.netlify.app/',
    codigo: 'https://github.com/GamaDev-Andres/BarberApp-fe',
    poster:
      'https://res.cloudinary.com/dapa84kxy/image/upload/v1644874961/portafolio/barberapp/barberapp_sdpgwl.png',
    fotos: []
  },
  {
    titulo: 'NETBOOK',
    descripcion:
      'Mini clon de una red social realizado con React, Redux y firebase, me baso en ciertas funcionalidades de Facebook, tales como, publicaciones, likes, comentarios y perfiles individuales.',
    tecnologias: ['React', 'Firebase', 'Redux', 'Sass'],
    url: 'https://agnetbook.netlify.app/',
    codigo: 'https://github.com/GamaDev-Andres/network-social-react',
    poster:
      'https://res.cloudinary.com/dapa84kxy/image/upload/v1644874998/portafolio/netbook/netbook_gus84s.png',
    fotos: []
  },
  {
    titulo: 'GALERIA',
    descripcion:
      'Simulación de una galería web realizada con React y Firebase , la aplicación se basa en la posibilidad de crear álbumes y tomarse fotos directamente en el álbum deseado y también tiene la posibilidad de buscar imágenes en una API y guardarlas en un álbum.',
    tecnologias: ['React', 'Firebase', 'StyledComponents'],
    url: 'https://practica-galeria.web.app/',
    codigo: 'https://github.com/GamaDev-Andres/react-galeria',
    poster:
      'https://res.cloudinary.com/dapa84kxy/image/upload/v1644875042/portafolio/galeriareact/reactFirebase_xjiauw.png',
    fotos: []
  },
  {
    titulo: 'MIS PELIS',
    descripcion:
      'Proyecto de películas hecho con Javascript, Sass y webpack .En este proyecto me baso en un diseño de una pagina funcional, consulto la API IMDB que me provee información tanto de series como de películas con sus respectivos tráileres y sinopsis. Implemento buscadores y lista de favoritos.',
    tecnologias: ['Html', 'Sass', 'Webpack', 'Javascript'],
    url: 'https://afmispelisgi.netlify.app/',
    codigo: 'https://github.com/GamaDev-Andres/PeliculasOMDB',
    poster:
      'https://res.cloudinary.com/dapa84kxy/image/upload/v1644875064/portafolio/mispelis/imagepelisOMDB_cchblf.png',
    fotos: []
  },
  {
    titulo: 'FINDYOURGAMES',
    descripcion:
      'Proyecto realizado con Html, css, javascript y gulp. En este proyecto consulto una API de videojuegos y organizo su información en una interfaz que me permite hacer búsqueda y filtrado de los videojuegos disponibles.',
    tecnologias: ['Html', 'Sass', 'Gulp', 'Javascript'],
    url: 'https://findyourgames.netlify.app/',
    codigo: 'https://github.com/GamaDev-Andres/FindYourGames',
    poster:
      'https://res.cloudinary.com/dapa84kxy/image/upload/v1644875081/portafolio/findyourgames/proyecto1_vsncjy.png',
    fotos: []
  }
]

export const miDescripcion = {
  quienSoy: {
    parrafo1: `Mi nombre es Andres Gama, me apasiona encontrar grandes retos que me
          permitan crecer profesional y personalmente, la programación me
          permite enfocarme en esto.Actualmente tengo 21 años y sé que lograre
          que mis conocimientos sean participes de grandes, útiles y reconocidos
          proyectos.`,
    parrafo2: `En mí podras ver a un chico ansioso por aprender y dispuesto a
          afrontar complejas curvas de aprendizaje y gracias a mi constancia y
          disciplina veras a alguien creciendo constantemente.`
  },
  proceso: {
    parrafo1: `Inicie con pequeños cursos y blogs de programación en YouTube,
            posteriormente me di cuenta de que este mundo de verdad me
            interesaba y a pesar de que lo miraba complicado ... debía
            intentarlo entonces me aventuré a profundizar y empecé a conocer
            plataformas como Udemy, Platzi, MDN, etc...`,
    parrafo2: `He tenido diferentes dificultades en mi proceso, tanto para aprender
            como para aplicar lo aprendido, entre estas dificultades se destaca
            el hecho de enfrentarme a retos desconocidos como pueden ser
            tecnologias nuevas o profundizar más una tecnología para obtener el
            resultado deseado o uno más optimo, al final entendí que lo más
            lindo de esta profesión es que se garantiza un crecimiento
            constante.`
  },
  presentacion:
    'apasionado por el mundo del desarrollo y por crear aplicaciones e interfaces que aporten y brinden grandes experiencias.'
}

export const links = {
  contacto: [
    {
      href: 'https://drive.google.com/file/d/1IIUutqCGZL_KSN7NqE-JdRYQl_6Qaqhd/view',
      type: 'Resumé'
    }
  ]
}
