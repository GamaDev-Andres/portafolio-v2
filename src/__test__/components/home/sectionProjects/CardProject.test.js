/**
 * @jest-environment jsdom
 */
import Enzyme from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import CardProject from '../../../../components/home/sectionProjects/CardProject'
import { proyectos } from '../../../../helpers/data'

describe('probando <CardProject/>', () => {
  const proyectoSinDetails = proyectos.at(-1)
  const wrapper = Enzyme.mount(
    <MemoryRouter>
      <CardProject project={proyectoSinDetails} />
    </MemoryRouter>
  )
  test('debe montarse correctamente', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('TitleAndLinksProject').length).toBe(1)
    expect(wrapper.find('ListOfTecnologias').exists()).toBe(true)
  })
  test('No debe montarse "Link" cuando no hay details', () => {
    expect(wrapper.find('Link').exists()).toBe(false)
  })
  test(' debe montarse "Link" cuando si hay details', () => {
    const proyectoConDetails = proyectos[0]
    const wrapper = Enzyme.mount(
      <MemoryRouter>
        <CardProject project={proyectoConDetails} withDetails={true} />
      </MemoryRouter>
    )
    expect(wrapper.find('Link').exists()).toBe(true)
  })
})
