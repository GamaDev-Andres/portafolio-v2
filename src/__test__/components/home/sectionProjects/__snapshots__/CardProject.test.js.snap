// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando <CardProject/> debe montarse correctamente 1`] = `
<div
  className=" shadow-blue-200 shadow-sm rounded-2xl rounded-t-none transition-colors duration-150 bg-white dark:bg-baseDark text-fondos"
>
  <a
    className="w-full block aspect-video"
    href="https://findyourgames.netlify.app/"
    rel="noreferrer"
    target="_blank"
  >
    <img
      alt="project"
      className="w-full aspect-video object-cover"
      src="https://res.cloudinary.com/dapa84kxy/image/upload/v1644875081/portafolio/findyourgames/proyecto1_vsncjy.png"
    />
  </a>
  <div
    className="px-4 pb-4"
  >
    <div
      className="relative py-2 flex justify-center items-center"
    >
      <a
        className="font-bold text-blue-400 text-center mx-auto text-xl"
        href="https://findyourgames.netlify.app/"
        rel="noreferrer"
        target="_blank"
      >
        FINDYOURGAMES
         
        <i
          className="text-sm fa-solid fa-link dark:text-letraDark text-fondos"
        />
      </a>
      <a
        className="absolute top-2 right-0 text-xl block text-letra"
        href="https://github.com/GamaDev-Andres/FindYourGames"
        rel="noreferrer"
        target="_blank"
      >
        <i
          className="dark:text-letraDark fa-brands fa-github"
        />
      </a>
    </div>
    <div>
      <h3
        className=" dark:text-letraDark text-letra font-medium"
      >
        Descripción
      </h3>
      <p>
        Proyecto realizado con Html, css, javascript y gulp. En este proyecto consulto una API de videojuegos y organizo su información en una interfaz que me permite hacer búsqueda y filtrado de los videojuegos disponibles.
      </p>
    </div>
    <div>
      <h3
        className="text-letra font-medium dark:text-letraDark"
      >
        Tecnologias implementadas
      </h3>
      <ul
        className="flex gap-2 flex-wrap py-2"
      >
        <li
          className="px-2 py-1 border border-bordes rounded-full"
          key="Html"
        >
          <span>
            Html
          </span>
        </li>
        <li
          className="px-2 py-1 border border-bordes rounded-full"
          key="Sass"
        >
          <span>
            Sass
          </span>
        </li>
        <li
          className="px-2 py-1 border border-bordes rounded-full"
          key="Gulp"
        >
          <span>
            Gulp
          </span>
        </li>
        <li
          className="px-2 py-1 border border-bordes rounded-full"
          key="Javascript"
        >
          <span>
            Javascript
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
